books/models.py

```diff
@@ -387,6 +387,22 @@ def get_subject_meta(self):
         APIField('subject_search_description')
     ]
 
+class k12SubjectBooks(models.Model):
+    subject = models.ForeignKey(snippets.k12Subject, on_delete=models.SET_NULL, null=True, related_name='k12subjects_subject')
```

Comment:
since you are using a snippet to manage these, I think you can forego the constants you added in `book.constants` and look below on line 407 at `BookCategory` to see how you can work with snippets on a page.